Tools/clinic/clinic.py

```diff
@@ -5212,10 +5321,33 @@ def parse_converter(
                     "Annotations must be either a name, a function call, or a string."
                 )
 
+    def parse_deprecated_positional(self, thenceforth: str) -> None:
+        assert isinstance(self.function, Function)
+        fname = self.function.full_name
+
+        if "." not in thenceforth:
+            fail(
+                f"Function {fname!r}: '* [from ...]' format expected to be "
+                f"'<major.minor>', where 'major' and 'minor' are digits."
+            )
```

Comment:
It is not actually needed.

Reply:
Yes, this should be caught in the try...except we introduced.